### PR TITLE
Clarify contribution for `main-distribution` packages

### PR DESCRIPTION
**Pull request by jryans**
This extends the contribution guide for `main-distribution` packages to describe two separate workflows

1. Release / snapshot
2. In-place build

each of which needs slightly different commands.

### CODE REVIEW

File: pkgs/racket-build-guide/contribute.scrbl
Comment (mflatt) [inline pkgs/racket-build-guide/contribute.scrbl:88]: "edit the package's catalog" is not quite right here. This should be something like "adjust the package installation to use the source specified by the main package catalog".

**Review by mflatt — CHANGES_REQUESTED** [review]
Thanks for this improvement! I have one request for a wording change.

File: pkgs/racket-build-guide/contribute.scrbl
Comment (jryans) [reply pkgs/racket-build-guide/contribute.scrbl]: Ah thanks, that's a better description of what's happening. 😄 I have updated this to match your wording.

**Comment by jryans** [comment]
@mflatt Thanks for taking a look, review comment addressed. 🙂 Someone will need to re-approve the workflows if a fresh CI pass is desired before merging.

**Pull request merged**
